lib/gst/plugin/gstclappersink.c

```diff
@@ -54,6 +56,29 @@ G_DEFINE_TYPE_WITH_CODE (GstClapperSink, gst_clapper_sink, GST_TYPE_VIDEO_SINK,
 GST_ELEMENT_REGISTER_DEFINE (clappersink, "clappersink", GST_RANK_NONE,
     GST_TYPE_CLAPPER_SINK);
 
+#define GST_TYPE_CLAPPER_SINK_ROTATION (gst_clapper_sink_rotation_get_type())
+static GType
+gst_clapper_sink_rotation_get_type (void)
+{
+  static GType rotation_type = 0;
+
+  static const GEnumValue rotations[] = {
+    {GST_VIDEO_ORIENTATION_AUTO, "Auto", "automatic"},
+    {GST_VIDEO_ORIENTATION_IDENTITY, "Original", "none"},
+    {GST_VIDEO_ORIENTATION_90R, "90°", "clockwise"},
+    {GST_VIDEO_ORIENTATION_180, "180°", "rotate-180"},
+    {GST_VIDEO_ORIENTATION_90L, "270°", "counterclockwise"},
+    {0, NULL, NULL},
+  };
+
+  if (!rotation_type) {
+    rotation_type = g_enum_register_static ("GstClapperSinkRotation",
+        rotations);
+  }
+  return rotation_type;
+}
+
+
```

Comment:
This seems unnecessary, considering a `GST_TYPE_VIDEO_ORIENTATION_METHOD` type already exists since GStreamer 1.20. Maybe we should just use that for consistency in values? It should be (I hope) relatively easy to implement remaining ones with `gsk_transform_rotate_3d` probably? And do 180 rotation around X/Y axis for flip? Or am I mistaken here? 🤔